flux-tests/tests/lib/range.rs

```diff
@@ -33,7 +33,12 @@ impl Rng {
 impl Iterator for RngIter {
     type Item = i32;
 
-    #[flux::sig(fn(self: &mut RngIter[@lo, @hi]) -> Option<i32{v:lo <= v && v < hi}>)]
+    #[flux::sig(fn(self: &strg RngIter[@in]) 
+                -> Option<i32{nextV: in.lo <= nextV && nextV < in.hi 
+                                  && nextV == in.cur}> 
+                ensures self: RngIter{out: out.lo == in.lo && out.hi == in.hi 
+                                        && out.cur == (in.cur + 1) || out.cur == in.hi})
+     ]
```

Comment:
I don't know what the standards are for formatting these signatures so I gave it my best go.

Reply:
Sweet! Can you think of a new client that can be verified using this more precise spec? Then you can add a new test in `flux-tests/tests/pos/surface/iter01` that exercises the spec? Perhaps something like:

```
pub fn test_next() {
    let mut rng = RngIter::new(0, 10);
    let x0 = rng.next();
    let x1 = rng.next();
    let x2 = rng.next();
    assert ( x0 == 0 && x1 == 1 && x2 == 2);
}
```

Reply:
also wondering if it's not enough to type the output like `Option<i32[in.curr]>`. If not I would like to understand why.

Reply:
`Option<i32[in.curr]>` works. That did not cross my mind.